mit-bitcoin-expo/mit-bitcoin-expo-2021/defending-against-99-attack.md

```diff
@@ -0,0 +1,76 @@
+
+And then I have this meme about hashes. 
+
+## Conclusion 
+Thanks! I've put this out as an idea. I'm not like, hey let's decrease the secuirty of all these things, but I do think it's an interesting topic. Are there areas here where we're leaving performance on the table? Or are we paradoxically reducing security by going too far? Questions, comments, disagreements? 
+
+I want to reiterate something that Neha said at the end of the last presentation. I work at the DCI, if you're interested in this kind of stuff get in touch. Get in contact we're pretty cool. Everything is remote now, so wherever you are in the world it doesn't make much of a difference. Are there any questoins?
+
+# Audience Questions
+Moderator: One of our questions is, would you suggest saving the state of the blockchain in some way to reduce resource consumption?
+
+Tadge: I didn't go into this in enough detail here but the idea of the utreexo project is this accumulator I'm working on with a bunch of open source contributors. It's sort of like pruning to whole UTXO set. In Bitcoin you can store the whole blockchain which is 256 gigs, or you can prune it and say look once I have the block I don't really need to keep it around. And I can store the UTXO set which is about 4 gigs, and is the current state of who owns what. And this basically prunes that as well. Don't store the UTXO set, store a little merkle root of it. It's not quite a merkle root, but it's very small, it's less than a kilobyte. And then transactions can prove that the UTXO they're spending exist with these merkle proofs. That's one example of where we can potentially leverage smaller proofs. It is a sort of different way of pruning the UTXO set. You don't need the blockchain, you don't need to UTXO set, you can run a full node in a very small amount of space.
+
+Moderator: we have time for about two more questions, the next question is: Converse to what you were saying about reducing the hash size, isn't increasing the hash size supposed to future proof the system instead?
+Tadge: Well, maybe. But you can't. You could say, let's put something new in Bitcoin where we have 512 bit pubkey hashes. Or we use an elliptic curve that 400 or 500 bits long, so we double the security. Well okay, but we still have almost everyone using the previous parts of the system. If somebody makes a computer that can derive a private key from a public key, which takes 2^128 ellitic curve ops. Maybe you're safe, but everyone else isn't, so the whole system isn't. If you're going to make your Bitcoins twice as secure, it doesn't matter, if everybody else's coins have been stolen it doesn't matter because the system has collapsed. It's no use to say "I still have mine", because everybody just leaves the system. So there may be ways to upgrade, but you kind of need everyone to upgrade, or almost everyone. If there's 5 million of the 21 million coins are stealable, the system is in trouble. 
```

Comment:
```suggestion
Tadge: Well, maybe. But you can't. You could say, let's put something new in Bitcoin where we have 512 bit pubkey hashes. Or we use an elliptic curve that 400 or 500 bits long, so we double the security. Well okay, but we still have almost everyone using the previous parts of the system. If somebody makes a computer that can derive a private key from a public key, which takes 2^128 elliptic curve ops. Maybe you're safe, but everyone else isn't, so the whole system isn't. If you're going to make your Bitcoins twice as secure, it doesn't matter, if everybody else's coins have been stolen it doesn't matter because the system has collapsed. It's no use to say "I still have mine", because everybody just leaves the system. So there may be ways to upgrade, but you kind of need everyone to upgrade, or almost everyone. If there's 5 million of the 21 million coins are stealable, the system is in trouble. 
```